spot_wrapper/wrapper.py

```diff
@@ -411,63 +411,61 @@ def wrapper_try_claim(self, *args, **kwargs):
     return wrapper_try_claim
 
 
+@dataclass()
+class SpotWrapperConfiguration:
+    """
+    Convenience class to specify the configuration of the spot wrapper
+
+    Attributes:
+        username: Username for authentication with the robot
+        password: Password for authentication with the robot
+        hostname: ip address or hostname of the robot
+        robot_name: Optional name of the robot
+        start_estop: If true, the wrapper will be an estop endpoint
+        estop_timeout: Timeout for the estop in seconds. The SDK will check in with the wrapper at a rate of
+                       estop_timeout/3 and if there is no communication the robot will execute a gentle stop.
+        rates: Dictionary of rates to apply when retrieving various data from the robot # TODO this should be an object to be unambiguous
+        callbacks: Dictionary of callbacks which should be called when certain data is retrieved # TODO this should be an object to be unambiguous
+        use_take_lease: Use take instead of acquire to get leases. This will forcefully take the lease from any
+                        other lease owner.
+        get_lease_on_action: If true, attempt to acquire a lease when performing an action which requires a
+                             lease. Otherwise, the user must manually take the lease. This will also attempt to
+                             power on the robot for commands which require it - stand, rollover, self-right.
+        continually_try_stand: If the robot expects to be standing and is not, command a stand.  This can result
+                               in strange behavior if you use the wrapper and tablet together.
+        rgb_cameras: If the robot has only body-cameras with greyscale images, this must be set to false.
+    """
+
+    username: str
+    password: str
+    hostname: str
+    robot_name: str
+    start_estop: bool = True
+    estop_timeout: float = 9.0
+    rates: typing.Optional[typing.Dict[str, float]] = None
+    callbacks: typing.Optional[typing.Dict[str, typing.Callable]] = None
+    use_take_lease: bool = False
+    get_lease_on_action: bool = False
+    continually_try_stand: bool = True
+    rgb_cameras: bool = True
+    payload_credentials_file: str = None
+    frame_prefix: str = field(init=False)
+
+    def __post_init__(self) -> None:
```

Comment:
For consistency with the old behavior, `__post_init__` does this also need to handle these cases from the original?

```
self._rates = rates or {}
self._callbacks = callbacks or {}
```

Otherwise I believe all of the `.get` calls below on `rates` and `callbacks` will fail for `NoneType` in all of the different tasks.

Or maybe just initialize them to empty lists instead of `None`?

Reply:
I'm not sure if dataclasses work in the same way as mutable initialisation arguments, if they do that would cause some issues. I've initialised in post-init.